### PR TITLE
add number of reversions option

### PR DESCRIPTION
**Pull request by stevemao**
Ref: https://github.com/sindresorhus/vinyl-paths/issues/2
Only thing is its probably not the best use-case to demonstrate in the README.md. However I can't use double `dest` because the `base` would change in after the first `dest()`. https://github.com/wearefractal/vinyl-fs/pull/77

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 var through = require('through2');
 
-module.exports = function () {
+module.exports = function (reversions) {
+	reversions = reversions || 1;
 	return through.obj(function (file, enc, cb) {
 		var history = file.history;
 
-		if (history.length > 1) {
+		if (history.length > reversions) {
```
Comment (sindresorhus) [inline index.js]: I don't think it should just silently not do its thing when there aren't enough history. Either throw or reverse as much as possible. Not sure which.

Comment (stevemao) [inline index.js]: reverse as much as possible? Since it's backward compatible any no one raised a request on throwing an error before.

**Comment by stevemao** [comment]
wearefractal/vinyl-fs#77 is merged and the added tests https://github.com/wearefractal/vinyl-fs/pull/79 passed. We could use double `dest` in the example (which makes more sense than the current one.).

File: index.js (hunk 1 above)
Comment (stevemao) [inline index.js]: Done. I think it makes more sense now

Comment (sindresorhus) [inline index.js]: > reverse as much as possible?

:+1:

**Comment by sindresorhus** [comment]
> We could use double dest in the example (which makes more sense than the current one.).

:+1:

File: index.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
 'use strict';
 var through = require('through2');
 
-module.exports = function () {
+module.exports = function (reversions) {
+	reversions = reversions || 1;
```
Comment (sindresorhus) [inline index.js]: `reversions = typeof reversions === 'number' ? reversions : 1;` as users might want to explicitly pass in `0` for some reason.

Comment (sindresorhus) [inline index.js]: should also have an empty line below this.

Comment (stevemao) [inline index.js]: very true

**Comment by stevemao** [comment]
double dest example needs to wait until `vinyl-fs` is released.

**Comment by sindresorhus** [comment]
Looks good. Thanks :)

**Pull request closed without merging**